python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -264,42 +352,92 @@ def _create_streaming_chat_message_content(
+            if settings.extra_parameters is not None and settings.extra_parameters.get("n", 1) > 1:
+                # Currently only OpenAI models allow multiple completions but the Azure AI Inference service
+                # does not expose the functionality directly. If users want to have more than 1 responses, they
+                # need to configure `extra_parameters` with a key of "n" and a value greater than 1.
+                raise ServiceInvalidExecutionSettingsError(
+                    "Auto invocation of tool calls may only be used with a single completion."
+                )
 
-            # If it's a user message and there are any image items in the message, we need to create a list of
-            # content items, otherwise we need to just pass in the content as a string or it will error.
-            contentItems = []
-            for item in message.items:
-                if isinstance(item, TextContent):
-                    contentItems.append(TextContentItem(text=item.text))
-                elif isinstance(item, ImageContent) and (item.data_uri or item.uri):
-                    contentItems.append(
-                        ImageContentItem(
-                            image_url=ImageUrl(url=item.data_uri or str(item.uri), detail=ImageDetailLevel.Auto)
-                        )
-                    )
-                else:
-                    logger.warning(
-                        "Unsupported item type in User message while formatting chat history for Azure AI"
-                        f" Inference: {type(item)}"
-                    )
-            chat_request_messages.append(_MESSAGE_CONVERTER[message.role](content=contentItems))
+    def _configure_function_choice_behavior(
+        self, settings: AzureAIInferenceChatPromptExecutionSettings, kernel: Kernel
+    ):
+        """Configure the function choice behavior to include the kernel functions."""
+
+        def _config_call_back(
+            function_choice_configuration: FunctionCallChoiceConfiguration,
+            settings: AzureAIInferenceChatPromptExecutionSettings,
+            type: str,
+        ):
+            """Update the settings from a FunctionChoiceConfiguration."""
+            if function_choice_configuration.available_functions:
+                settings.tool_choice = type
+                # The list of tool objects will be initialized with the JSON string returned by
+                # `kernel_function_metadata_to_function_call_format`.
+                settings.tools = [
+                    kernel_function_metadata_to_function_call_format(f)
+                    for f in function_choice_configuration.available_functions
+                ]
+
+        settings.function_choice_behavior.configure(
+            kernel=kernel, update_settings_callback=_config_call_back, settings=settings
+        )
 
-        return chat_request_messages
+    async def _process_function_calls(
```

Comment:
is this not generic as well?

Reply:
Yes, a similar function exists in the OpenAI connector but it's used differently.